### PR TITLE
IE should prefer bound props over static props other than class

### PR DESCRIPTION
**Pull request by fergaldoyle**
A follow on from: https://github.com/vuejs/vue/pull/1639

There are more attributes other than `title` which IE has issues with. (`id`, `lang`, `language`,`dir`, etc)
e.g: http://codepen.io/anon/pen/yYxBPy

In this PR, we feature detect binding syntax issues, if the current browser does have this issue, we check every prop for a bound alternative, (`:`, `v-bind:`, `*.sync`, `*.once`) and if present, use the bound prop over a static prop.

`class` is ignored. Buy ignoring it the behavior matches other browsers (bound classes get passed down to component root element)

### CODE REVIEW

File: src/compiler/compile-props.js [hunk 1]
```diff
@@ -8,6 +8,15 @@ var empty = {}
 var identRE = require('../parsers/path').identRE
 var settablePathRE = /^[A-Za-z_$][\w$]*(\.[A-Za-z_$][\w$]*|\[[^\[\]]+\])*$/
 
+// feature detect browsers (IE) that have trouble
+// with binding syntax on certain attributes
+var div, preferBinding
```
Comment (yyx990803) [inline src/compiler/compile-props.js]: Let's move this check in to `util/env.js` and expose it as `_.preferBinding`.

File: src/compiler/compile-props.js [hunk 2]
```diff
@@ -120,6 +125,28 @@ module.exports = function compileProps (el, propOptions) {
...
+ * @return {String} attr
+ */
+
+function checkBindingAttr (el, attr) {
+  if (attr === 'class') {
+    return null
+  }
+
+  return (
+    el.getAttribute(':' + attr) ||
```
Comment (yyx990803) [inline src/compiler/compile-props.js]: Looks like we can just use `el.hasAttribute` here and the returned value will be a boolean.

File: src/compiler/compile-props.js [hunk 3]
```diff
@@ -120,6 +125,28 @@ module.exports = function compileProps (el, propOptions) {
...
+    el.getAttribute(':' + attr + '.once') ||
+    el.getAttribute(':' + attr + '.sync') ||
```
Comment (yyx990803) [inline src/compiler/compile-props.js]: Is the check for `.once` and `.sync` necessary? Does IE also treat `:title.once` as `title`?

**Comment by fergaldoyle** [comment]
`hasAttribute` now used and indeed there is no need to check for `.once` or `.sync`

**Pull request merged**